src/LuigisBoxBundle/ValueObject/SearchUrlBuilder.php

```diff
@@ -0,0 +1,319 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Answear\LuigisBoxBundle\ValueObject;
+
+use Webmozart\Assert\Assert;
+
+/**
+ * @see https://live.luigisbox.com/#search-as-a-service-tips
+ */
+class SearchUrlBuilder
+{
+    private const AVAILABLE_ORDER_DIRECTIONS = ['asc', 'desc'];
+    public const ARRAY_ITEM_SEPARATOR = ':';
+    public const LIST_SEPARATOR = ',';
+    public const DEFAULT_SIZE = 10;
+
+    /**
+     * @var string|null
+     */
+    private $q;
+
+    /**
+     * @var bool|null
+     */
+    private $qu;
```

Comment:
Given there's a method to transform properties into query I'd use more descriptive names